insights/parsers/firewall_config.py

```diff
@@ -0,0 +1,30 @@
+"""
+FirewallDConf - file ``/etc/firewalld/firewalld.conf``
+======================================================
+
+The FirewallDConf class parses the file ``/etc/firewalld/firewalld.conf``.
+And returns a dict contains the firewall configurations.
+
+Examples:
+
+    >>> type(firewalld)
+    <class 'insights.parsers.firewall_config.FirewallDConf'>
+    >>> 'DefaultZone' in firewalld
+    True
+    >>> firewalld['DefaultZone']
+    'public'
+"""
+
+from insights.specs import Specs
+from insights import Parser, parser
+from insights.parsers import split_kv_pairs
+
+
+@parser(Specs.firewalld_conf)
+class FirewallDConf(Parser, dict):
+    """
+    Class for parsing ``/etc/firewalld/firewalld.conf`` file.
+    """
+
+    def parse_content(self, content):
+        self.update(split_kv_pairs(content, use_partition=False))
```

Comment:
Better add an `Empty check` at the end of this function and raise a `SkipException` for `empty content`, by doing so, there will be no parser result when the content is empty, which is good for component dependency check.